tests/integration/operatorhub/cmd_service_test.go

```diff
@@ -29,459 +29,467 @@ var _ = Describe("odo service command tests for OperatorHub", func() {
-			if os.Getenv("KUBERNETES") == "true" {
-				Skip("This is a OpenShift specific scenario, skipping")
-			}
-
-			componentName := helper.RandString(6)
-			helper.CmdShouldPass("odo", "create", "nodejs", componentName)
-			helper.CmdShouldPass("odo", "push")
-
-			stdOut := helper.CmdShouldFail("odo", "link", "EtcdCluster/example")
-			Expect(stdOut).To(ContainSubstring("Couldn't find service named %q", "EtcdCluster/example"))
-		})
-
-		It("should successfully connect and disconnect a component with an existing service", func() {
-			if os.Getenv("KUBERNETES") == "true" {
-				Skip("This is a OpenShift specific scenario, skipping")
-			}
-
-			componentName := helper.RandString(6)
-			helper.CmdShouldPass("odo", "create", "nodejs", componentName)
-			helper.CmdShouldPass("odo", "push")
-
-			// start the Operator backed service first
-			operators := helper.CmdShouldPass("odo", "catalog", "list", "services")
-			etcdOperator := regexp.MustCompile(`etcdoperator\.*[a-z][0-9]\.[0-9]\.[0-9]-clusterwide`).FindString(operators)
-			helper.CmdShouldPass("odo", "service", "create", fmt.Sprintf("%s/EtcdCluster", etcdOperator), "--project", commonVar.Project)
-			helper.CmdShouldPass("odo", "push")
-
-			// now verify if the pods for the operator have started
-			pods := oc.GetAllPodsInNs(commonVar.Project)
-			// Look for pod with example name because that's the name etcd will give to the pods.
-			etcdPod := regexp.MustCompile(`example-.[a-z0-9]*`).FindString(pods)
-
-			ocArgs := []string{"get", "pods", etcdPod, "-o", "template=\"{{.status.phase}}\"", "-n", commonVar.Project}
-			helper.WaitForCmdOut("oc", ocArgs, 1, true, func(output string) bool {
-				return strings.Contains(output, "Running")
-			})
-
-			stdOut := helper.CmdShouldPass("odo", "link", "EtcdCluster/example")
-			Expect(stdOut).To(ContainSubstring("Successfully created link between component"))
-			helper.CmdShouldPass("odo", "push")
-			stdOut = helper.CmdShouldFail("odo", "link", "EtcdCluster/example")
-			Expect(stdOut).To(ContainSubstring("already linked with the service"))
-
-			// Before running "odo unlink" checks, wait for the pod to come up from "odo push" done after "odo link"
-			pods = oc.GetAllPodsInNs(commonVar.Project)
-			componentPod := regexp.MustCompile(fmt.Sprintf(`%s-.[a-z0-9]*-.[a-z0-9\-]*`, componentName)).FindString(pods)
-			ocArgs = []string{"get", "pods", componentPod, "-o", "template=\"{{.status.phase}}\"", "-n", commonVar.Project}
-			helper.WaitForCmdOut("oc", ocArgs, 1, true, func(output string) bool {
-				return strings.Contains(output, "Running")
-			})
-
-			stdOut = helper.CmdShouldPass("odo", "unlink", "EtcdCluster/example")
-			Expect(stdOut).To(ContainSubstring("Successfully unlinked component"))
-			helper.CmdShouldPass("odo", "push")
-
-			// verify that sbr is deleted
-			stdOut = helper.CmdShouldFail("odo", "unlink", "EtcdCluster/example")
-			Expect(stdOut).To(ContainSubstring("failed to unlink the service"))
 		})
 	})
 })
```

Comment:
I think there are a few test cases missing from this PR. See https://github.com/openshift/odo/blob/main/tests/integration/operatorhub/cmd_service_test.go
But I guess you must be aware of those since those are your changes :sweat_smile:

Reply:
right, I just rebased on main and added the missing tests